proposals/0387-cross-compilation-destinations.md

```diff
@@ -167,75 +180,174 @@ that, a separate `toolset.json` file is introduced:
+version of this proposal, hence `"schemaVersion": "3.0"`. We'll keep parsing `"version": 1` and `"version": 2` for
+backward compatibility, but for consistency with `info.json` this field is renamed to `"schemaVersion"`. Here's an
+informally defined schema for these files:
+
+```json5
+{
+  "schemaVersion": "3.0",
+  "runTimeTriples": [
+    "<triple1>": {
+      "sdkRootPath": "<a required path relative to `destination.json` containing SDK root>",
+      // all of the properties listed below are optional:
+      "swiftResourcesPath": "<a path relative to `destination.json` containing Swift resources for dynamic linking>",
+      "swiftStaticResourcesPath": "<a path relative to `destination.json` containing Swift resources for static linking>",
+      "includeSearchPaths": ["<array of paths relative to `destination.json` containing headers>"],
+      "librarySearchPaths": ["<array of paths relative to `destination.json` containing libraries>"],
+      "toolsetPaths": ["<array of paths relative to `destination.json` containing toolset files>"]
+    },
+    // a destination can support more than one run-time triple:
+    "<triple2>": {
+      "sdkRootPath": "<a required path relative to `destination.json` containing SDK root>",
+      // all of the properties listed below are optional:
+      "swiftResourcesPath": "<a path relative to `destination.json` containing Swift resources for dynamic linking>",
+      "swiftStaticResourcesPath": "<a path relative to `destination.json` containing Swift resources for static linking>",
+      "includeSearchPaths": ["<array of paths relative to `destination.json` containing headers>"],
+      "librarySearchPaths": ["<array of paths relative to `destination.json` containing libraries>"],
+      "toolsetPaths": ["<array of paths relative to `destination.json` containing toolset files>"]
+    }
+    // more triples can be supported by a single destination if needed, primarily for sharing files between them.
+  ]
+}
+```
+
+We propose that all relative paths in `destination.json` files should be validated not to "escape" the destination
+bundle for security reasons, in the same way that `toolset.json` files are validated when contained in destination
+bundles. That is, `../` components, if present in paths, will not be allowed to reference files and
+directories outside of a corresponding destination bundle. Symlinks will also be validated to prevent them from escaping
+out of the bundle.
+ 
+If `sdkRootPath` is specified and `swiftResourcesPath` is not, the latter is inferred to be
+`"\(sdkRootPath)/usr/lib/swift"` when linking the Swift standard library dynamically, `"swiftStaticResourcesPath"` is
+inferred to be `"\(sdkRootPath)/usr/lib/swift_static"` when linking it dynamically. Similarly, `includeSearchPaths` is
+inferred as `["\(sdkRootPath)/usr/include"]`, `librarySearchPaths` as  `["\(sdkRootPath)/usr/lib"]`.
```

Comment:
Isn't this changing the current behavior, ie these `-I`/`-L` flags are not added by default to the SDK path? Maybe these two include/lib paths should be kept empty by default.

Reply:
We won't populate them in SwiftPM explicitly with `-I`/`-L` when these properties are absent or empty in `destination.json`, but that's my understanding of what the behavior is when only `-sdk` option is passed to `swiftc`.

Reply:
> that's my understanding of what the behavior is when only -sdk option is passed to swiftc.

No, it doesn't on linux, see the output from apple/swift#63416. The reason is that the SDK is the C sysroot and these are Swift include/lib flags, so the compiler only looks in the Swift resource directory, not the normal non-Swift SDK locations you list here.

Reply:
I think if you update this doc to mention that nothing is done by default, should be okay.